src/main/java/com/ecore/roles/configuration/RestTemplateConfiguration.java

```diff
@@ -1,14 +1,33 @@
 package com.ecore.roles.configuration;
 
+import lombok.RequiredArgsConstructor;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.web.client.HttpServerErrorException;
 import org.springframework.web.client.RestTemplate;
 
 @Configuration
+@RequiredArgsConstructor
 public class RestTemplateConfiguration {
+    private final RetryConfigurationProperties retryConfigurationProperties;
+
     @Bean
     public RestTemplate restTemplate(RestTemplateBuilder builder) {
         return builder.build();
     }
+
+    @Bean
+    public RetryTemplate retryTemplate() {
```

Comment:
We can improve client's calls in order to be a fault tolerant services with spring-retry for 5XX http status response errors